src/ir/intrinsics.cpp

```diff
@@ -45,4 +46,58 @@ Call* Intrinsics::isCallWithoutEffects(Expression* curr) {
   return nullptr;
 }
 
+bool Intrinsics::isConfigureAll(Function* func) {
+  return func->module == JSPrototypesModule && func->base == ConfigureAll;
+}
+
+Call* Intrinsics::isConfigureAll(Expression* curr) {
+  if (auto* call = curr->dynCast<Call>()) {
+    if (auto* func = module.getFunctionOrNull(call->target)) {
+      if (isConfigureAll(func)) {
+        return call;
+      }
+    }
+  }
+  return nullptr;
+}
+
+std::vector<Name> Intrinsics::getConfigureAllFunctions(Call* call) {
+  assert(isConfigureAll(call));
+
+  auto error = [&](const char* msg) {
+    Fatal() << "Invalid configureAll( " << msg << "): " << *call;
+  };
```

Comment:
We should definitely avoid erroring out on unrecognized `configureAll` patterns, since they would still be perfectly valid. Maybe we can print a warning and return an empty list if we see some other pattern.

Reply:
I think it might be better to error for now? A warning might be missed, and make things harder to debug. (We can always reduce an error to a warning later if it feels annoying in practice.)